src/main/kotlin/com/ampnet/web3provider/service/impl/AccountInformationServiceImpl.kt

```diff
@@ -0,0 +1,57 @@
+package com.ampnet.web3provider.service.impl
+
+import com.ampnet.web3provider.controller.pojo.JsonRpcRequest
+import com.ampnet.web3provider.controller.pojo.ProviderResponse
+import com.ampnet.web3provider.enums.RedisEntity
+import com.ampnet.web3provider.repository.RedisRepository
+import com.ampnet.web3provider.service.AccountInformationService
+import com.ampnet.web3provider.service.DefaultProviderService
+import mu.KLogging
+import org.springframework.stereotype.Service
+
+@Service
+class AccountInformationServiceImpl(
+    private val redisRepository: RedisRepository,
+    private val defaultProviderService: DefaultProviderService
+) : AccountInformationService {
+
+    companion object : KLogging()
+
+    override fun getBalance(request: JsonRpcRequest): ProviderResponse {
+        val address = request.params[0]
+        val blockParameter = request.params[1]
+        logger().info { "Received request to get eth_getBalance for address: $address and block: $blockParameter" }
+        val hashKey = address + blockParameter
+        redisRepository.getCache(RedisEntity.BALANCE.methodName, hashKey)?.let { return ProviderResponse(request, it) }
```

Comment:
Code here can be extracted into a private method to avoid duplication:
```kotlin
private fun getResponseFromCacheOrProvider(entity: RedisEntity, hashKey: String, request: JsonRpcRequest): ProviderResponse {
    redisRepository.getCache(entity.methodName, hashKey)?.let { return ProviderResponse(request, it) }
    val providerResponse = defaultProviderService.getResponse(request)
    redisRepository.updateCache(entity, hashKey, providerResponse.result)
    return providerResponse
}
```